src/framework/components/anim/controller.js

```diff
@@ -533,7 +888,7 @@ Object.assign(AnimController.prototype, {
             if (transition.time > 0) {
                 clip.blendWeight = 0.0;
             } else {
-                clip.blendWeight = 1.0 / activeState.totalWeight;
+                clip.blendWeight = activeState.animations[i].weight / activeState.totalWeight;
```

Comment:
is it possible for totalWeight to be `0` here?

Reply:
Hmm yeah direct blending with weights that cancel each other out. I'll update this to check for potential divide by zero cases.